WORKFLOW.md

```diff
@@ -0,0 +1,50 @@
+_below is more like open disscussion than a design doc_
+### Introduction
+
+The following CRDs are defined for managing the lifecycle of Buckets:
+
+ - BucketRequest - Represents a request to provision a Bucket
+ - BucketClass - Represents a class of Buckets with similar characteristics
+ - Bucket - Represents a Bucket or its equivalent in the storage backend
+
+ The following CRDs are defined for managing the lifecycle of workloads accessing the Bucket:
+
+ - BucketAccessRequest - Represents a request to access a Bucket
+ - BucketAccessClass - Represents a class of accessors with similar access requirements
+ - BucketAccess - Represents a access token or service account in the storage backend
+
+Simialr to CSI, it has three running components :
+- Controller - mananges life cycle of COSI objects, watches for the requests for above mentioned objects
+- Node adapter - bootstraps bucket access request to applications for their worloads
+- Sidecar/Driver - receives the request and calls corresponding api in the driver
+
+More details and defintions can be found the [COSI Spec](https://github.com/kubernetes-sigs/container-object-storage-interface-spec/blob/master/spec.md)
+
+### Workflow
+
+- BucketRequest(namespace scoped) can be created with reference to a BucketClass, so user will get a Bucket(clusterscoped). The COSI driver will create a `backend bucket` in RGW for a Bucket Request
```

Comment:
why do we need to create an actual bucket in RGW in this case?
if no one would access this bucket, it could be left at a CRD level.
this would solve the issue of deciding which user creates and own the bucket?